tests/components/netatmo/test_sensor.py

```diff
@@ -0,0 +1,233 @@
+async def test_process_battery(model, data, expected):
+    """Test battery level translation."""
+    assert sensor.process_battery(data, model) == expected
+
+
+@pytest.mark.parametrize(
+    "angle, expected",
+    [
+        (0, "N"),
+        (40, "NE"),
+        (70, "E"),
+        (130, "SE"),
+        (160, "S"),
+        (220, "SW"),
+        (250, "W"),
+        (310, "NW"),
+        (340, "N"),
+    ],
+)
+async def test_process_angle(angle, expected):
+    """Test wind direction translation."""
+    assert sensor.process_angle(angle) == expected
+
+
+@pytest.mark.parametrize(
+    "angle, expected",
+    [(-1, 359), (-40, 320)],
+)
+async def test_fix_angle(angle, expected):
+    """Test wind angle fix."""
+    assert sensor.fix_angle(angle) == expected
+
+
+@pytest.mark.parametrize(
+    "uid, name, expected",
+    [
+        ("12:34:56:37:11:ca-reachable", "netatmo_mystation_reachable", "True"),
+        ("12:34:56:03:1b:e4-rf_status", "netatmo_mystation_yard_radio", "Full"),
+        (
+            "12:34:56:05:25:6e-rf_status",
+            "netatmo_valley_road_rain_gauge_radio",
+            "Medium",
+        ),
+        (
+            "12:34:56:36:fc:de-rf_status_lvl",
+            "netatmo_mystation_netatmooutdoor_radio_level",
+            "65",
+        ),
+        (
+            "12:34:56:37:11:ca-wifi_status_lvl",
+            "netatmo_mystation_wifi_level",
+            "45",
+        ),
+        (
+            "12:34:56:37:11:ca-wifi_status",
+            "netatmo_mystation_wifi_status",
+            "Full",
+        ),
+        (
+            "12:34:56:37:11:ca-temp_trend",
+            "netatmo_mystation_temperature_trend",
+            "stable",
+        ),
+        (
+            "12:34:56:37:11:ca-pressure_trend",
+            "netatmo_mystation_pressure_trend",
+            "down",
+        ),
+        ("12:34:56:05:51:20-sum_rain_1", "netatmo_mystation_yard_rain_last_hour", "0"),
+        ("12:34:56:05:51:20-sum_rain_24", "netatmo_mystation_yard_rain_today", "0"),
+        ("12:34:56:03:1b:e4-windangle", "netatmo_mystation_garden_direction", "SW"),
+        (
+            "12:34:56:03:1b:e4-windangle_value",
+            "netatmo_mystation_garden_angle",
+            "217",
+        ),
+        ("12:34:56:03:1b:e4-gustangle", "mystation_garden_gust_direction", "S"),
+        (
+            "12:34:56:03:1b:e4-gustangle",
+            "netatmo_mystation_garden_gust_direction",
+            "S",
+        ),
+        (
+            "12:34:56:03:1b:e4-gustangle_value",
+            "netatmo_mystation_garden_gust_angle_value",
+            "206",
+        ),
+        (
+            "12:34:56:03:1b:e4-guststrength",
+            "netatmo_mystation_garden_gust_strength",
+            "9",
+        ),
+        (
+            "12:34:56:26:68:92-health_idx",
+            "netatmo_baby_bedroom_health",
+            "Fine",
+        ),
+    ],
+)
+async def test_weather_sensor_enabled(hass, config_entry, uid, name, expected):
```

Comment:
The sensor types include wifi signal strength and so on, it makes the test name a bit confusing I think,

Also, enabling a disabled entity will not happen until after a cooldown period has expired, are you sure this works as intended?
You should first check the sensor state is not there, then after that enable the entity and check again it is there.

Maybe something like this: https://github.com/home-assistant/core/pull/48090/files

Reply:
Not sure if the updated test addresses exactly what you mean.